### PR TITLE
Issue#4select energizer from list

### LINKED ISSUE #4: As a host, I want to select an energiser from a list
- create frontend REACT components for the list of energizers,
- create endpoint that fetches data from the database and serves it to frontend 
- populate database with a selection of energizers

### PR DESCRIPTION
**Pull request by RobCso**
## Naming Rules

#4-As a host I want to select an energiser from a list- Robert_Csonka

## Description

I have created a simple endpoint "/api/energizers" that accesses all the data from the energizers database, I  have created a very simple component that fetches the data from the endpoint and displays the name and description of an energizer

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #4 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Review by JustinasBrazauskas — APPROVED** [review]
Good job!

File: server/api.js [hunk 1]
```diff
@@ -2,11 +2,40 @@ import { Router } from "express";
...
+		.then(() => {
+			res
+				.status(201)
+				.json({ message: "The energizer was succesfully uploaded" });
+		})
+		.catch((error) => {
+			res.status(500).send(error);
+		});
+});
+
```
Comment (RobCso) [inline server/api.js:42]: I have only created this POST endpoint to update my database, its very incomplete as it lacks any validation, but it works and can be used as a base

File: client/src/pages/Energisers.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import Navbar from "./components/Navbar";
+import Footer from "./components/Footer.js";
+import EnergisersCard from "./components/EnergisersCard.js";
+import "./Energisers.css";
+// import NumberOfPages from "./components/NumberOfPages";
```
Comment (bertilnilsson) [inline client/src/pages/Energisers.js]: Why is there commented out code?

File: client/src/pages/Home.js [hunk 3]
```diff
@@ -2,26 +2,27 @@ import Navbar from "./components/Navbar";
 import Footer from "./components/Footer";
 import FavesCard from "./components/FavesCard";
 import WhatsNewCard from "./components/WhatsNewCard";
+
 import "./Home.css";
 
 export function Home() {
 	return (
-	<main className="main-page">
```
Comment (bertilnilsson) [inline client/src/pages/Home.js:9]: Just a general thought - have you agreed style guidelines and you're therefore changing the white space (on this and other pages). 

The white space should remain consistency so that each person commiting code is not changing it back and forth and generating unnecessary changes.

File: server/api.js (hunk 1 above)
Comment (bertilnilsson) [reply server/api.js:42]: Maybe make a comment in the code about the missing validation.

File: server/api.js [hunk 4]
```diff
@@ -2,11 +2,40 @@ import { Router } from "express";
...
 const router = Router();
 
 router.get("/", (_, res) => {
 	logger.debug("Welcoming everyone...");
 	res.json({ message: "Hello, world!" });
 });
 
+router.get("/energizers", async (_, res) => {
+	logger.debug("Welcoming everyone...");
```
Comment (bertilnilsson) [inline server/api.js:15]: Don't think this is necessary, if anything you could be logging the output of the query to debug?

File: server/api.js [hunk 5]
```diff
@@ -2,11 +2,40 @@ import { Router } from "express";
...
+	const query = "select * from energizers";
```
Comment (bertilnilsson) [inline server/api.js]: So I would say it's not very good practice to return every column from a database directly to the client, for now it would work but as you add columns they would get automatically included here which might not be intended.

Also minor point by SQL reserved words (SELECT, FROM) are generally always written uppercase.

**Review by bertilnilsson — COMMENTED** [review]
I've made some comments, nothing major and could be merged if it's blocking other people's work but worth considering now or for future PRs.

File: client/src/pages/Energisers.js (hunk 2 above)
Comment (AzinYad) [reply client/src/pages/Energisers.js]: I have made a component for NumberOfPages which is some Icon if we needed to go to next page I will add that component and css style to Robert work if it was needed

File: client/src/pages/Home.js (hunk 3 above)
Comment (AzinYad) [reply client/src/pages/Home.js:9]: yes we did , its the style that related to every page on this App

Comment (bertilnilsson) [reply client/src/pages/Home.js:9]: I'm not commenting on the CSS - I'm talking about the white space change that is generating this whole diff.

File: client/src/pages/Energisers.js (hunk 2 above)
Comment (bertilnilsson) [reply client/src/pages/Energisers.js]: But it's not part of this code

File: client/src/pages/Home.js (hunk 3 above)
Comment (AzinYad) [reply client/src/pages/Home.js:9]: I can edit the css if u are agree with
As we agreed in team once we had a meeting

Comment (bertilnilsson) [reply client/src/pages/Home.js:9]: I'm not talking about the CSS. I'm only talking about white space. If you don't understand just ignore, it doesn't matter for the code for now it's just about how to effectively manage PRs.

Comment (AzinYad) [reply client/src/pages/Home.js:9]: O I I got It now , sorry 
yes u are right

File: client/src/pages/Energisers.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import { useState, useEffect } from "react";
+
+function Energisers() {
+	const [allEnergizers, setAllEnergizers] = useState([]);
+	useEffect(() => {
+		fetch("/api/energizers")
```
Comment (AzinYad) [inline client/src/pages/Energisers.js:11]: I was wondering for an hour trying to find  bug , because it wasn't working for me , and it was the spell of energizer , because my end point was /energiser so , I don't mind changing all them in my code to Z but we have to agree on something and stick to it in team ,because it might get complicated later as we carry on codding

File: server/api.js (hunk 1 above)
Comment (RobCso) [reply server/api.js:42]: added the comments

File: client/src/pages/Home.js (hunk 3 above)
Comment (RobCso) [reply client/src/pages/Home.js:9]: I have not thought about the white space issue but understand why it could be a problem , I have also remade my code after taking on board Azin's suggestions which were valid, and that might have contributed to the white space issue I'll try to keep that in mind in the future

File: server/api.js (hunk 4 above)
Comment (RobCso) [reply server/api.js:15]: I had no exposure to loger.debug and was just following the existing code to make sure my code works

File: server/api.js (hunk 5 above)
Comment (RobCso) [reply server/api.js]: Thank you, good points, I have updated the SQL reserved words to uppercase I will try sticking to that convention, regarding the return of every column its only what I have been exposed to during the course, what do you suggest would be the better way to return from database

File: client/src/pages/Energisers.js (hunk 6 above)
Comment (RobCso) [reply client/src/pages/Energisers.js:11]: I think we should spell with Z as its more prevalent on the internet and even VSCODE highlights the s spelling as unknown word

Comment (AzinYad) [reply client/src/pages/Energisers.js:11]: so I will change in my code and we should also edit page names
